src/platforms/PolygonId/components/PolygonIdQRCode.tsx

```diff
@@ -0,0 +1,130 @@
+import {
+  Box,
+  Center,
+  HStack,
+  IconButton,
+  ModalBody,
+  ModalHeader,
+  Spinner,
+  Text,
+  useBreakpointValue,
+} from "@chakra-ui/react"
+import useGuild from "components/[guild]/hooks/useGuild"
+import useUser from "components/[guild]/hooks/useUser"
+import Button from "components/common/Button"
+import ErrorAlert from "components/common/ErrorAlert"
+import { SignedValdation, useSubmitWithSign } from "hooks/useSubmit"
+import { ArrowLeft, ArrowsClockwise } from "phosphor-react"
+import { QRCodeSVG } from "qrcode.react"
+import { useEffect, useState } from "react"
+import useSWR, { mutate } from "swr"
+import { Role } from "types"
+import fetcher from "utils/fetcher"
+
+type Props = {
+  role: Role
+  isClaimed: boolean
+  goBack: () => void
+}
+
+const PolygonIdQRCode = ({ role, goBack, isClaimed }: Props) => {
+  const { id: userId } = useUser()
+  const { id: guildId } = useGuild()
+  const [shouldFetchQR, setShouldFetchQR] = useState(null)
+  const QR_URL = `${process.env.NEXT_PUBLIC_POLYGONID_API}/v1/users/${userId}/polygon-id/claim/${guildId}:${role.id}/qrcode`
+
+  const submit = async (signedValidation: SignedValdation) =>
+    fetcher(`${process.env.NEXT_PUBLIC_POLYGONID_API}/v1/polygon-id/claim`, {
+      method: "POST",
+      ...signedValidation,
+    })
+
+  const claim = useSubmitWithSign(submit, {
+    onSuccess: () => setShouldFetchQR(QR_URL),
+  })
+  const qr = useSWR(shouldFetchQR)
+
+  const qrSize = useBreakpointValue({ base: 300, md: 400 })
+
+  useEffect(() => {
+    if (userId && guildId && !isClaimed) {
+      claim.onSubmit({
```

Comment:
This onSubmit should happen conditionally when the user presses the "Mint proof" button on a `MintableRole` card. So in short, I think we should check if there's a QR code for a role and if we can fetch it, we just display that QR in a modal when the user clicks on the "Show QR code" button. If there's no QR code, the button's label is "Mint proof" and if the user clicks on it, we fire this onSubmit & also open the modal in a loading state.